### DISCUSSION TITLE
Computing Current Density Over an Interface

### DISCUSSION THREAD

**Discussion by jsevic** [idaholab/moose#20280] [Q&A General] [answered]
I have a 2-D model of a thin-film dielectric stack with top and bottom ohmic contacts and I wish to compute the total current over either contact. Do I need to define an auxiliary variable, say J(x,y,t), from E(x,y,t) * sigma_electrical(x,y,t) and then perform an integration over either one of the contacts, say over x for fixed y? Is the integration something I can do in a post-processing step or would it too be an auxiliary variable? 

Thank you!

John Sevic

**Comment by GiudGiud** [comment, marked as answer]
Hello

A total current seems like the job for a postprocessor. 
Whether you start from J = E * sigma or whether you provide E and sigma for the integration depends on your preferences. I suspect if you use J you ll be able to use an existing SideIntegration postprocessor

Guillaume

**Comment by jsevic** [comment]
Ok cool let me try thx!

John Sevic

**Comment by jsevic** [comment]
Another question related to this.

I recall on an earlier simulation I did, I used an auxiliary kernel to compute electrical conductivity based on the product mobility and charge density and electron charge.

If I define electrical conductivity presently using a ParsedMaterial, and call this sigma_e, is there a way to pass this sigma_e to the Exodus output so I can plot it? I used AuxKernal for this before.

Thx!

**Reply by GiudGiud** [reply]
From a ParsedMaterial you can output to exodus with this
https://mooseframework.inl.gov/source/auxkernels/MaterialRealAux.html

**Reply by GiudGiud** [reply]
or you can specify the `output` parameter of the parsedmaterial to the exodus output

**Comment by jsevic** [comment]
Ok I will try I also tried AuxVariable and AuxKernel and this seemed to work too.
